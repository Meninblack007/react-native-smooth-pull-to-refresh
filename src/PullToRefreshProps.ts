import {NativeScrollEvent, NativeSyntheticEvent} from "react-native";

export interface PullToRefreshProps {
    isRefreshing: boolean;
    onRefresh: () => void;
    contentComponent: JSX.Element;
    minPullDistance?: number;
    backgroundColor?: string;
    onScroll?: (event: NativeSyntheticEvent<NativeScrollEvent>) => void;
    onTriggerToRefresh?: (isRefresh: boolean) => void;
    isReachEnd?: boolean;
    toPosition?: number;
}

export const defaultMinPullDistance = 120;
export const defaultPTRBackgroundColor = "#f6f6f6";
export const pullAnimatedBackgroundColor = "#f6f6f6";
