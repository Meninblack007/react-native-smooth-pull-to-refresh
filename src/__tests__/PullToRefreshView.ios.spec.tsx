import {shallow} from "enzyme";
import React from "react";
import {ScrollView} from "react-native";
import sinon from "sinon";
import {PullToRefreshProps} from "../PullToRefreshProps";
import {PullToRefreshView} from "../PullToRefreshView.ios";

describe("PullToRefreshView ios spec", () => {
  it("App show PullToRefreshView ios", () => {
    const stub = sinon.stub();
    const props: PullToRefreshProps = {
      isRefreshing: false,
      onRefresh: stub,
      contentComponent: <ScrollView />,
      pullAnimHeight: 70,
      pullAnimYValues: {from: -50, to: 10},
    };
    const wrapper = shallow(<PullToRefreshView {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
