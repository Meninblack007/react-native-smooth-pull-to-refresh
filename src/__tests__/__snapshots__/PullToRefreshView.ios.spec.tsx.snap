// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullToRefreshView ios spec App show PullToRefreshView ios 1`] = `
<withState(withState(withHandlers(lifecycle(BaseComponent))))
  contentComponent={<ScrollView />}
  isRefreshing={false}
  onRefresh={[Function]}
  pullAnimHeight={70}
  pullAnimYValues={
    Object {
      "from": -50,
      "to": 10,
    }
  }
  setShouldTriggerRefresh={[Function]}
  shouldTriggerRefresh={false}
/>
`;
